pkg/git_provider/gitlab.go

```diff
@@ -0,0 +1,356 @@
+package git_provider
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/quickube/piper/pkg/conf"
+	"github.com/quickube/piper/pkg/utils"
+
+	"github.com/xanzy/go-gitlab"
+)
+
+type GitlabClientImpl struct {
+	client *gitlab.Client
+	cfg    *conf.GlobalConfig
+}
+
+func NewGitlabClient(cfg *conf.GlobalConfig) (Client, error) {
+	ctx := context.Background()
+
+	client, err := gitlab.NewClient(cfg.GitProviderConfig.Token)
+	if err != nil {
+		log.Fatalf("Failed to create client: %v", err)
+	}
+
+	err = ValidateGitlabPermissions(ctx, client, cfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to validate permissions: %v", err)
+	}
+
+	group, resp, err := client.Groups.GetGroup(cfg.OrgName, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get organization: %v", err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("failed to get organization data %s", resp.Status)
+	}
+
+	cfg.OrgID = int64(group.ID)
+
+	log.Printf("Org ID is: %d\n", cfg.OrgID)
+
+	return &GitlabClientImpl{
+		client: client,
+		cfg:    cfg,
+	}, err
+}
+
+func (c *GitlabClientImpl) ListFiles(ctx *context.Context, repo string, branch string, path string) ([]string, error) {
```

Comment:
Write tests
here is an setup for that:
https://github.com/xanzy/go-gitlab/blob/main/gitlab_test.go#L41